test/e2e/channel_chain_test.go

```diff
@@ -0,0 +1,131 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+/*
+TestChannelChain tests the following scenario:
+
+EventSource ---> Channel ---> Subscriptions ---> Channel ---> Subscriptions ---> Service(Logger)
+
+*/
+func TestChannelChain(t *testing.T) {
+	clients, cleaner := Setup(t, t.Logf)
+
+	const (
+		senderName = "e2e-channelchain-sender"
+		routeName  = "e2e-channelchain-route"
+	)
+	var channelNames = [2]string{"e2e-channelchain1", "e2e-channelchain2"}
+	var subscriptionNames1 = CreateRandomSubscriptionNames("e2e-complexscen-subs1")
```

Comment:
Could we add some comments about why we are adding randomness to the tests? In general I prefer tests to be repeatable so just curious why it's a good idea or what it tests?

Reply:
Same as above.